File: cli/src/commands/registry.ts
import { defineCommand } from "citty";
import { addRegistry, removeRegistry, readConfig } from "../config.ts";

const registryAdd = defineCommand({
  meta: { name: "add", description: "Save a named registry source for reuse (e.g. a team's own template repo)" },
  args: {
    name: { type: "positional", required: true, description: "Short name, e.g. \"acme\"" },
    source: {
      type: "positional",
      required: true,
      description: 'Local path or git source, e.g. "github:acme/hackpack-templates"',
    },
  },
  async run({ args }) {
    await addRegistry(args.name, args.source);
    console.log(`Saved registry "${args.name}" -> ${args.source}`);
    console.log(`Use it with: hackpack new --registry=${args.name}`);
  },
});

const registryRemove = defineCommand({
  meta: { name: "remove", description: "Remove a saved registry" },
  args: {
    name: { type: "positional", required: true },
  },
  async run({ args }) {
    await removeRegistry(args.name);
    console.log(`Removed registry "${args.name}"`);
  },
});

const registryList = defineCommand({
  meta: { name: "list", description: "List saved registries" },
  async run() {
    const config = await readConfig();
    const entries = Object.entries(config.registries);
    if (entries.length === 0) {
      console.log("No saved registries. Add one with: hackpack registry add <name> <source>");
      return;
    }
    for (const [name, source] of entries) console.log(`${name}  ->  ${source}`);
  },
});

export const registryCommand = defineCommand({
  meta: { name: "registry", description: "Manage template registry sources" },
  subCommands: { add: registryAdd, remove: registryRemove, list: registryList },
});

File: go-cli/cmd/hackpack/registry.go
package main

import (
	"fmt"

	"github.com/spf13/cobra"

	"hackpack/internal/config"
)

func newRegistryCmd() *cobra.Command {
	cmd := &cobra.Command{
		Use:   "registry",
		Short: "Manage template registry sources",
	}

	add := &cobra.Command{
		Use:   "add <name> <source>",
		Short: "Save a named registry source for reuse (e.g. a team's own template repo)",
		Args:  cobra.ExactArgs(2),
		RunE: func(cmd *cobra.Command, args []string) error {
			if err := config.AddRegistry(args[0], args[1]); err != nil {
				return err
			}
			fmt.Printf("Saved registry %q -> %s\n", args[0], args[1])
			fmt.Printf("Use it with: hackpack new --registry=%s\n", args[0])
			return nil
		},
	}

	remove := &cobra.Command{
		Use:   "remove <name>",
		Short: "Remove a saved registry",
		Args:  cobra.ExactArgs(1),
		RunE: func(cmd *cobra.Command, args []string) error {
			if err := config.RemoveRegistry(args[0]); err != nil {
				return err
			}
			fmt.Printf("Removed registry %q\n", args[0])
			return nil
		},
	}

	list := &cobra.Command{
		Use:   "list",
		Short: "List saved registries",
		Args:  cobra.NoArgs,
		RunE: func(cmd *cobra.Command, args []string) error {
			cfg, err := config.Read()
			if err != nil {
				return err
			}
			if len(cfg.Registries) == 0 {
				fmt.Println("No saved registries. Add one with: hackpack registry add <name> <source>")
				return nil
			}
			for name, source := range cfg.Registries {
				fmt.Printf("%s  ->  %s\n", name, source)
			}
			return nil
		},
	}

	cmd.AddCommand(add, remove, list)
	return cmd
}
